app/src/main/java/org/wikipedia/talk/TalkTopicsViewModel.kt

```diff
@@ -0,0 +1,78 @@
+package org.wikipedia.talk
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.MutableStateFlow
+import org.wikipedia.database.AppDatabase
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.dataclient.discussiontools.DiscussionToolsInfoResponse
+import org.wikipedia.dataclient.mwapi.MwQueryResponse
+import org.wikipedia.page.Namespace
+import org.wikipedia.page.PageTitle
+import org.wikipedia.staticdata.TalkAliasData
+import org.wikipedia.staticdata.UserTalkAliasData
+
+class TalkTopicsViewModel : ViewModel() {
+
+    private val talkPageSeenRepository = TalkPageSeenRepository(AppDatabase.instance.talkPageSeenDao())
+    private val handler = CoroutineExceptionHandler { _, throwable ->
+        _uiState.value = UiState.Error(throwable)
+    }
+
+    private lateinit var pageTitle: PageTitle
```

Comment:
This ViewModel should take a PageTitle as a factory parameter, similar to our previous ViewModels.

Reply:
Sure! Done.